src/Tinfoil/AEAD/AESGCM/Data.hs

```diff
@@ -34,7 +36,7 @@ newtype RandomField =
 -- startup.
 newtype InvocationCount =
   InvocationCount {
-    unInvocationCount :: Integer
+    unInvocationCount :: Word64
```

Comment:
Sorry, I see this is intentional from what you spec'd, it isn't quite what I expected, but I think I understand why. I think I would of checked and made sure it fit in 32 bits and left the type as Word32 here to make sure.